### DISCUSSION TITLE
Configuration to enable UPnP search

### DISCUSSION THREAD

**Discussion by edolis** [gerbera/gerbera#2690] [Q&A]
Sorry to bother but i am pretty lost on this.
Installed on Debian (Raspbian 10) version 1.1.0 and I am struggling to enable DLNA search function.
My final goal is simply being able to search the DLNA library from an Android app which can perform UPnP searches.

What I have tried so far:
In config.xml I have added a node `<upnp></upnp>` inside `<server>`
Result was this error 
`[src/server.cc:123] upnp_init(): upnp_init: UpnpInit failed
2022-08-15 18:27:45   ERROR: main: upnp error -203
2022-08-15 18:27:45   ERROR: Could not bind to socket.
2022-08-15 18:27:45    INFO: Please check if another instance of Gerbera or
2022-08-15 18:27:45    INFO: another application is running on port 0.
`
So I added a explicit port under the server node (49153, to debug) and now the error is gone.

Yet I am trying to test some search (using BubbleUPnP Android app) and any search yields "No search results"
Now is it possible to log the search requests/response? I see there are parameters for the upnp node - no clue which one could be "standard" to allow proper communication with Bubble.
Or even better, can someone suggest a configuration which makes search work? I have seen that in development phase Bubble and request/response were logged, but I'm too much user level to understand how this was done.
Found no example in the docs or in the web.
Please help.

Thanks!
note - The library is of course populated, I can browse albums from the same BubbleUPnP app, but cannot find them using the search function)

**Comment by KarlStraussberger** [comment]
Hi, if you really installed 1.1 you should try to get a more recent version (we're on 1.11 now). Especially searching was limited back then (I don't even remember).

In the current version there is a command line option `--create-config` which prints out a working default config that you can redirect to your `config.xml`.

**Comment by edolis** [comment]
Thanks!
The packages for Debian linked in https://docs.gerbera.io/en/stable/install.html are for the 1.1.0. (I think this should be mentioned in the installation docs of the site - I expected to be able to install a reasonably recent version using apt install)
I spent some time trying to understand how install from deb-multimedia, and the farthest I can reach without broken dependencies is the 1.3.2 (deb-multimedia buster).

The search function works but with issues - is there a way I can log request/response from the gerbera side to understand if the issue is on client side/config?
Is there some way I can install manually a version newer than 1.3.2 (can you please point me to some resource to try understanding how fix broken dependencies?)

Appreciate you patience.

**Reply by edolis** [reply]
(asking for way to install more recent versions than 1.3.2 since it looks like I cannot make Autoscan work, any folder gives the message _This is a persistent autoscan directory (defined in config), it can not be modified via the web UI._ even if there is no autoscan section in the config file. I can just add directory static (and not even understood how you can force a refresh manually). 
Also, 1.3.2 cannot complete the manual scan of folder. It restarts at a certain point of the scan. So it's not workable on Raspberry Pi 4.
I bet this is an issue already fixed, in a version I currently don't know how to install.)

**Reply by whyman** [reply]
If you follow the Ubuntu install instructions, the repo also contains Debian releases

```
curl -fsSL https://gerbera.jfrog.io/artifactory/api/gpg/key/public | sudo apt-key add -
sudo apt-add-repository https://gerbera.jfrog.io/artifactory/debian
```

Let us know how it goes.

**Reply by edolis** [reply]
thanks Whyman. Unfortunately does not seem to add flawlessly on Raspbian :(

`pi@raspi00:~ $ sudo apt-add-repository https://gerbera.jfrog.io/artifactory/debian
Traceback (most recent call last):
File "/usr/bin/apt-add-repository", line 95, in <module>
sp = SoftwareProperties(options=options)
 File "/usr/lib/python3/dist-packages/softwareproperties/SoftwareProperties.py", line 109, in __init__
    self.reload_sourceslist()
 File "/usr/lib/python3/dist-packages/softwareproperties/SoftwareProperties.py", line 599, in reload_sourceslist
    self.distro.get_sources(self.sourceslist)
 File "/usr/lib/python3/dist-packages/aptsources/distro.py", line 93, in get_sources
    (self.id, self.codename))
aptsources.distro.NoDistroTemplateException: Error: could not find a distribution template for Raspbian/buster
pi@raspi00:~ $
`

**Reply by whyman** [reply]
Hrm, does raspbian require some special setup in a debian repo? (I am not a debian user! :smile:)
